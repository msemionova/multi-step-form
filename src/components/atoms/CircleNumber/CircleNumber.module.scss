@import '../../../styles/main';

.circle {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 3.6rem;
  width: 3.6rem;
  border: .1rem solid $color-white;
  border-radius: $border-radius-large;
  color: $color-white;
  font-weight: 700;
  font-size: $default-font-size;
  line-height: 1;
}

.active {
  background-color: $color-tertiary;
  border-color: $color-tertiary;
  color: $color-primary-dark;
}
