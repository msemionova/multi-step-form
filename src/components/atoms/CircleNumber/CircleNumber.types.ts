export type CircleNumberProps = {
  children: React.ReactNode;
  isActive?: boolean;
}
