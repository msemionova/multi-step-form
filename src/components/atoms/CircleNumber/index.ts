export { default as CircleNumber } from './CircleNumber';
export type { CircleNumberProps } from './CircleNumber.types';
