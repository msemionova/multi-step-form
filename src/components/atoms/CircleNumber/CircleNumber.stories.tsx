import type { Meta, StoryObj } from '@storybook/react';

import CircleNumber from './CircleNumber';

const meta: Meta<typeof CircleNumber> = {
  component: CircleNumber,
};

export default meta;
type Story = StoryObj<typeof CircleNumber>;

export const Default: Story = {
  args: {
    children: '1'
  }
};

export const Active: Story = {
  args: {
    children: '1',
    isActive: true
  }
};
