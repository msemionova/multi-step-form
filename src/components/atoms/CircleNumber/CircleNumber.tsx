import styles from './CircleNumber.module.scss';
import { CircleNumberProps } from './CircleNumber.types';

export default function CircleNumber({ isActive = false, children }: CircleNumberProps) {
  const classes = [styles['circle'], isActive ? styles['active'] : ''].filter(Boolean).join(' ');
  return (
    <div className={classes}>
      {children}
    </div>
  );
}
