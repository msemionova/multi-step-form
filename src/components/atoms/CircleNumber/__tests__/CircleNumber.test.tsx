import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import CircleNumber from '../CircleNumber';

describe('<CircleNumber />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CircleNumber>Test</CircleNumber>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display the text passed as props', () => {
    const { getByText } = render(<CircleNumber>Test</CircleNumber>);
    expect(getByText('Test')).toBeInTheDocument();
  });
});
