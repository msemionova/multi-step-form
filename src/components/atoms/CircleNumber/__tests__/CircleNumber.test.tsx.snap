// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleNumber /> renders correctly 1`] = `
<div
  className="circle"
>
  Test
</div>
`;
